.circleci/config.yml

```diff
@@ -0,0 +1,200 @@
+      - run: 
+          name: Login DockerHub
+          command: echo "$DOCKERHUB_PASS" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
+  
+jobs:
+  format:
+    executor: test-executor
+    description: Format Check
+    steps:
+      - checkout
+      - rust_setup
+      - run:
+          name: Format check
+          command: cargo fmt -- --check
+  lint:
+    executor: test-executor
+    description: Lint Code
+    steps:
+      - build_setup
+      - run:
+          name: Clippy Check
+          command: cargo clippy --all-targets
+      - save_build_cache
+  test:
+    executor: test-executor
+    description: Run tests
+    steps:
+      - build_setup
+      - run:
+          name: Run Tests
+          command: cargo test --all
+      - save_build_cache
+  security:
+    executor: test-executor
+    description: Cargo audit
+    steps:
+      - checkout  
+      - rust_setup
+      - cargo_audit_setup
+      - run:
+          name: Check for known security issues in dependencies
+          command: cargo audit
+  docker-build:
+    executor: test-executor
+    description: Build Docker Image
+    steps:
+      - checkout
+      - run:
+          name: Build docker image
+          command: BUILDKIT_PROGRESS=plain DOCKER_BUILDKIT=1 docker build -t $IMAGE_NAME:latest .
+      - run:
+          name: Archive Docker image
+          command: docker save -o image.tar $IMAGE_NAME
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./image.tar
+  docker-publish-latest:
+    executor: test-executor
+    description: Publish latest Docker Image
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Load Archived Docker Image
+          command: docker load -i /tmp/workspace/image.tar
+      - login_dockerhub
+      - run:
+          name: Publish Latest Docker Image
+          command: docker push $IMAGE_NAME:latest
+  docker-publish-tag:
+    executor: test-executor
+    description: Publish Tagged Docker Image
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Load Archived Docker Image
+          command: docker load -i /tmp/workspace/image.tar
+      - login_dockerhub
+      - run:
+          name: Publish Tagged Docker Image
+          command: |
+            IMAGE_TAG=${CIRCLE_TAG/v/''}
+            docker tag $IMAGE_NAME:latest $IMAGE_NAME:$IMAGE_TAG
+            docker push $IMAGE_NAME:latest
+            docker push $IMAGE_NAME:$IMAGE_TAG
+
+workflows:
+  test-code:
+    jobs:
+      - format
+      - lint
+      - test
+  security:
+    jobs:
+      - security
+  docker:
+    jobs:
+      - docker-build
+      - docker-publish-latest:
+          requires:
+            - docker-build
+          filters:
+            branches:
+              only: main
+      - docker-publish-tag:
+          requires:
+            - docker-publish-latest
+          filters:
+            tags:
+              only: /^v.*/
```

Comment:
What is this tag for?

Reply:
This will trigger publish `chainsafe/pint:x.x.x` images everytime we tagging PINT with `x.x.x`